IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -421,26 +426,83 @@ - (BOOL)saveToken:(__unused MSIDCredentialCacheItem *)credential
 {
     assert(credential);
     assert(serializer);
+    
     [self updateLastModifiedForCredential:credential context:context];
-    NSData *itemData = [serializer serializeCredentialCacheItem:credential];
+    
+    if (key.isShared)
+    {
+        return [self saveSharedCredential:credential key:key context:context error:error];
+    }
+    else
+    {
+        return [self saveAppCredential:credential key:key context:context error:error];
+    }
+}
 
+- (BOOL)saveAppCredential:(__unused MSIDCredentialCacheItem *)credential
+                      key:(__unused MSIDCacheKey *)key
+                  context:(__unused id<MSIDRequestContext>)context
+                    error:(__unused NSError **)error
+{
+    assert(credential);
+    MSIDCacheItemJsonSerializer *serializer = [[MSIDCacheItemJsonSerializer alloc] init];
```

Comment:
nit: create one and reuse for the whole class

Reply:
Done.